v0/destinations/branch/transform.js

```diff
@@ -83,25 +63,51 @@ function getCategoryAndName(rudderEventName) {
 function getUserData(message) {
   const { context } = message;
   const { os } = message.context;
-
   if (!os || !isDefinedAndNotNullAndNotEmpty(os.name)) {
     throw new CustomError(
       "os name is missing in the payload and please make sure to insert it at context.os.name",
       400
     );
   }
+  const userData = {};
 
-  return removeUndefinedAndNullValues({
-    os: os.name,
-    os_version: os.version,
-    app_version: context.app.version,
-    screen_dpi: context.screen.density,
-    android_id: get(context, "android_id") ? context.android_id : null,
-    idfa: get(context, "idfa") ? context.android_id : null,
-    idfv: get(context, "idfv") ? context.android_id : null,
-    aaid: get(context, "aaid") ? context.android_id : null,
-    developer_identity: getFieldValueFromMessage(message, "userId")
-  });
+  userData.os = os.name;
+  userData.os_version = os.version;
+  userData.app_version = context.app.version;
+  userData.screen_dpi = context.screen.density;
+  userData.developer_identity = getFieldValueFromMessage(message, "userId");
+
+  if (isDefinedAndNotNull(os.name)) {
```

Comment:
As we are already checking `os` and `os.name` in line 66, we can omit this check.

Reply:
Okay